### PR TITLE
fix(DB): Terenthis quest chain

### LINKED ISSUE #2118: Quest Event missing complete A Lost Master & How Big a Threat & Npc Sentinel Selarin [$25]
##### SMALL DESCRIPTION:

To make it a little easier here are all my sources and scripts I found

https://github.com/Anonymous2/trinityone/commit/65aa7cd57bda49ab7f9ea3f0efefc81d5d0a8f3b

https://github.com/ElunaLuaEngine/ElunaTrinityWotlk/commit/8edb6e77abff1a9b20df226f14b7a81b9e1acefc

https://github.com/ElunaLuaEngine/ElunaTrinityWotlk/commit/05b55109710ef0d20a7b8e41d405c066e0d432be

https://github.com/TrinityCore/TrinityCore/issues?utf8=%E2%9C%93&q=Terenthis

**Here is a description of how the quest should work** 

How Big a Threat
https://vanilla-twinhead.twinstar.cz/?issue=10624

https://github.com/TrinityCore/TrinityCore/issues/14364

**After the Quest How Big a Threat**

A Lost Master

https://vanilla-twinhead.twinstar.cz/?issue=10623
https://vanilla-twinhead.twinstar.cz/?issue=3419

After the Quest A Lost **Master**

**Escape Through Force**

If you do the quest Escape Through Force and you have completed the escort, the master should hold a small talk with his bear and then despawn both. 

Escape Through Stealth or Escape Through Force

The order of the quest would be

1 Quest 984 How Big a Threat Part 1
2 Quest 985 How Big a Threat Part 2
3 Quest 986 A Lost Master Part 1
4 Quest 993 A Lost Master Part 2
5 Quest 994 or 995 Escape Through Stealth or Escape Through Force

After delivering the quest to Terenthis, the Sentinel Selarin should run down the path in house and keep with Terenthis Small-talk then it should stand for 3 minutes and give a quest.

After the 3 minutes it goes again and despawnt in the woods

##### EXPECTED BLIZZLIKE BEHAVIOUR:

**So here are the videos from Offi and from Ac Core**

https://youtu.be/hXFftFeQvb4?t=617 Quest How Big a Threat Part 1 Offi

https://youtu.be/dLmm15NnYLk Quest How Big a Threat Part 1 Ac Core

https://youtu.be/hXFftFeQvb4?t=1152 Quest How Big a Threat Part 2 Offi

https://youtu.be/-262Z3tsjVk Quest How Big a Threat Part 2 Ac Core

https://youtu.be/hXFftFeQvb4?t=1188 Quest A Lost Master Part 1 Offi

https://youtu.be/vZZp8Hr3y8M Quest A Lost Master Part 1 Ac Core

https://youtu.be/hXFftFeQvb4?t=11347 Quest A Lost Master Part 2 Offi

https://youtu.be/LvtcwRzESGg Quest A Lost Master Part 2 Ac Core

https://youtu.be/hXFftFeQvb4?t=13262 Magic of the cape Quest A Lost Master Offi

https://youtu.be/LvtcwRzESGg?t=22 Magic of the cape Quest A Lost Master Ac Core

https://youtu.be/hXFftFeQvb4?t=13286 Quest Escape Through Force Offi

https://youtu.be/y1vwYhbqaiA Quest Escape Through Force Ac Core

https://youtu.be/hXFftFeQvb4?t=13685 Npc Sentinel Selarin run Offi

https://youtu.be/-JbQaZ2TzUs  Npc Sentinel Selarin run Ac Core

to find the quest order here well

https://vanilla-twinhead.twinstar.cz/?quest=985

##### CURRENT BEHAVIOUR:

It doesn't happen after the quest's submissions no Event.

##### STEPS TO REPRODUCE THE PROBLEM:
.go c 37097
accept the quest here
and then give the quest back.

there are 5 quests in total
1 Quest How Big a Threat Part 1
2 Quest How Big a Threat Part 2
3 Quest A Lost Master Part 1
4 Quest A Lost Master Part 2
5 Quest Escape Through Force

##### BRANCH(ES):
Master

##### AC HASH/COMMIT:
AzerothCore rev. 7b105b75dd2a+ 2019-07-15 06:08:55 +0000 (master branch) (Win64, Release)
##### OPERATING SYSTEM:
Windows 10

##### MODULES:
mod-transmog
mod-world-chat

<bountysource-plugin>

---
There is a **[$25 open bounty](https://www.bountysource.com/issues/77369019-quest-event-missing-complete-a-lost-master-how-big-a-threat-npc-sentinel-selarin?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by claudiodfc**
## Changes Proposed:
- Implemented Set Orientation on waypoint pause by reading the waypoint orientation from DB
- Imported from trinitycore and vmangos data
- Fixed Volcor's gossip and quest deliver
- Fixed Sentinel Selarin's pathing and flags
- Scripted events for quests:
Quest 984 How Big a Threat Part 1
Quest 985 How Big a Threat Part 2
Quest 986 A Lost Master Part 1
Quest 993 A Lost Master Part 2
Quest 994 or 995 Escape Through Stealth or Escape Through Force

## Issues Addressed:

- Closes #2118

## SOURCE:

Check issue for Source

## Tests Performed:

- Tested in windows 10

## How to Test the Changes:

1. .go c id 3693
2. Be around level 15 and do the quest chain checking for the RP interactions displayed on the issue

## Known Issues and TODO List:

None

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by IntelligentQuantum — APPROVED** [review]

**Comment by claudiodfc** [comment]
@azerothcore/testers can someone take a look at this?

**Comment by yuzifc** [comment]
Tested, working great :)

https://user-images.githubusercontent.com/70452524/148704628-ed9b5ddc-e009-4d15-ad0b-6880c814ea71.mp4

File: data/sql/updates/pending_db_world/rev_1639532770750121300.sql [hunk 1]
```diff
@@ -0,0 +1,160 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1639532770750121300');
+
+-- Senetil Selarin spawn after completition of quest=995 or quest=994
+UPDATE `creature_template` SET `speed_walk`=1 WHERE `entry`= 3694;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1639532770750121300.sql:4]: sniffed?

File: data/sql/updates/pending_db_world/rev_1639532770750121300.sql [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+(3695, 1, 0, '%s begins to moan and roar at Terenthis while stomping his paws on the ground.', 16, 0, 100, 53, 0, 474, 1228, 0, 'Grimclaw'),
+(3695, 2, 0, '%s roars at Terenthis more, but this time the druid seems to understand the bear.', 16, 0, 100, 0, 0, 473, 1231, 0, 'Grimclaw'),
+(3695, 3, 0, '%s roars in acknowledgement at Terenthis.', 16, 0, 100, 0, 0, 473, 1233, 0, 'Grimclaw'),
+(3695, 4, 0, '%s growls in your direction before taking time to sniff you.', 16, 7, 100, 0, 0, 0, 1234, 0, 'Grimclaw'),
+(3695, 5, 0, '%s faces southeast and whimpers before looking back at you.', 16, 7, 100, 0, 0, 0, 1225, 0, 'Grimclaw'),
+(3695, 6, 0, '%s roars in excitement as he rushes Volcor. When he reaches his master, Grimclaw licks his face.', 16, 7, 100, 0, 0, 0, 1241, 0, 'Grimclaw');
+
+-- Waypoints for Grimclaw and Sentinel Selarin
+DELETE FROM `waypoints` WHERE `entry` IN (3694, 3695, 369500);
+INSERT INTO `waypoints` (`entry`, `pointid`, `position_x`, `position_y`, `position_z`, `orientation`, `point_comment`) VALUES
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1639532770750121300.sql:107]: sniffed?

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1639532770750121300.sql (hunk 2 above)
Comment (claudiodfc) [reply data/sql/updates/pending_db_world/rev_1639532770750121300.sql:107]: Vmangos

File: data/sql/updates/pending_db_world/rev_1639532770750121300.sql (hunk 1 above)
Comment (claudiodfc) [reply data/sql/updates/pending_db_world/rev_1639532770750121300.sql:4]: Vmangos

**Comment by yuzifc** [comment]
tested, grimclaw looking the wrong way, sentinel the same thing.

https://user-images.githubusercontent.com/70452524/148708158-45a1556b-e28f-4df2-873c-33ee0c9be530.mp4

**Comment by claudiodfc** [comment]
Looks like a smartai problem. When the waypoint pauses it should either use the orientation from db or keep the current orientation from movement. This is a generic issue but I'll take a look later.

**Comment by claudiodfc** [comment]
@IntelligentQuantum Can you give another review on core change?

**Review by IntelligentQuantum — APPROVED** [review]

**Comment by temperrr** [comment]
Looks good!

**Comment by acidmanifesto** [comment]
@Kitzunu can you go over the re-review and approve if you find nothing wrong please? Since this deals with a quest chain and addresses a slight smart ai issue, i would like to see this go forward if there is no further issue identified.

**Review by acidmanifesto — APPROVED** [review]

**Comment by fendwe** [comment]
See also https://github.com/chromiecraft/chromiecraft/issues/3216

**Comment by claudiodfc** [comment]
The issue can't be this PR since I tested several times doing the full chain on several machines and no issues were found.

**Comment by Kosiak** [comment]
It's very hard to explain everything in details in here but ill try. You fixed the quest chain but u broke Grimclaw(3695). With scripts u summoning creature which already exists. And all scripts applied to the [Grimclaw](https://classic.wowhead.com/npc=3695/grimclaw) which is supposed to be at he's location.  So your quest is working because u summon him and do waypoints from very close to quest. But u broke original Grimclaw which supposed to be at his location all the time and at respawn aka server restart he constantly on the move which explained in this [PR](https://github.com/azerothcore/azerothcore-wotlk/issues/10925). But he's constantly on the move due to your scripts starts on respawn(exactly what happens after server restarts). U "fixed" it with scripts but they broke Grimclaw completely. And while your quest chain was working. [This](https://github.com/azerothcore/azerothcore-wotlk/pull/10926) pr however overruled your waypoints and all the quest chain in a big mess now. We need to find a way to separate original Grimclaw from the one that is summoned for quest. I hope it wasn't too long.

**Comment by claudiodfc** [comment]
I see.. I'll see what I can do about it. Thanks for the feedback.

**Pull request merged**